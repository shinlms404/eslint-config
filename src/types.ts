import type { Rules } from 'eslint-define-config'

export type AwaitAble<T> = T | Promise<T>

export type ConfigRules = Partial<Rules>

export type InteropDefault<T> = T extends { default: infer U } ? U : T
