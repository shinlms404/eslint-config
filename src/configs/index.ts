export * from './comments'
export * from './ignores'
export * from './imports'
export * from './javascript'
export * from './jsonc'
export * from './markdown'
export * from './prettier'
export * from './sort-keys'
export * from './typescript'
export * from './vue'
