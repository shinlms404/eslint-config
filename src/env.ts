import process from 'node:process'

export const isInEditor = !!(
  (process.env.VSCODE_PID ||
    process.env.VSCODE_CWD ||
    process.env.JETBRAINS_IDE ||
    process.env.VIM) &&
  !process.env.CI
)
