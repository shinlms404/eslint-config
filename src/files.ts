export const JS = '**/*.?([cm])js'
export const JSX = '**/*.?([cm])jsx'

export const TS = '**/*.?([cm])ts'
export const TSX = '**/*.?([cm])tsx'

export const JSON = '**/*.json'
export const JSON5 = '**/*.json5'
export const JSONC = '**/*.jsonc'

export const VUE = '**/*.vue'

export const NODE_MODULES = '**/node_modules'
export const DIST = '**/dist'
export const LOCKFILE = [
  '**/package-lock.json',
  '**/yarn.lock',
  '**/pnpm-lock.yaml',
  '**/bun.lockb'
]
export const EXCLUDE = [
  NODE_MODULES,
  DIST,
  ...LOCKFILE,

  '**/output',
  '**/coverage',
  '**/temp',
  '**/fixtures',
  '**/.vitepress/cache',
  '**/.nuxt',
  '**/.vercel',
  '**/.changeset',
  '**/.idea',
  '**/.output',
  '**/.vite-inspect',

  '**/CHANGELOG*.md',
  '**/*.min.*',
  '**/LICENSE*',
  '**/__snapshots__',
  '**/auto-import?(s).d.ts',
  '**/components.d.ts',

  'package.json'
]
