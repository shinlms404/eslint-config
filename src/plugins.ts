// @ts-nocheck

export * as pluginTs from '@typescript-eslint/eslint-plugin'
export * as configPrettier from 'eslint-config-prettier'
export * as pluginComments from 'eslint-plugin-eslint-comments'
export * as pluginImport from 'eslint-plugin-i'
export * as pluginJsonc from 'eslint-plugin-jsonc'
export * as pluginPrettier from 'eslint-plugin-prettier'
export * as pluginSortKeys from 'eslint-plugin-sort-keys'
export * as pluginUnicorn from 'eslint-plugin-unicorn'
export * as pluginUnusedImports from 'eslint-plugin-unused-imports'
export * as pluginVue from 'eslint-plugin-vue'

export * as parserTs from '@typescript-eslint/parser'
export * as parserJsonc from 'jsonc-eslint-parser'
export * as parserVue from 'vue-eslint-parser'
