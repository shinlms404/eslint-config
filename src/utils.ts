import type { FlatESLintConfig } from 'eslint-define-config'
import { getPackageInfoSync, isPackageExists } from 'local-pkg'
import type { AwaitAble, InteropDefault } from './types'

export async function flat(
  ...configs: AwaitAble<FlatESLintConfig[]>[]
): Promise<FlatESLintConfig[]> {
  const resolved = await Promise.all(configs)
  return resolved.flat()
}

export function getVueVersion(): number | undefined {
  if (isPackageExists('vue')) {
    const pkg = getPackageInfoSync('vue', { paths: [process.cwd()] })
    const version = pkg?.version?.split('.') ?? []
    if (version) {
      return Number(version[0])
    }
  }
}

export function interopDefault<T>(m: T): InteropDefault<T> {
  return (m as any).default || m
}
