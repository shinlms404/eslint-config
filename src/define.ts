import type { FlatESLintConfig } from 'eslint-define-config'
import {
  comments,
  ignores,
  imports,
  javascript,
  jsonc,
  prettier,
  sortKey,
  typescript,
  vue,
  yaml
} from './configs/index'
import type { AwaitAble } from './types'
import { flat } from './utils'

export async function defineESlintConfig(): Promise<FlatESLintConfig[]> {
  const configs: AwaitAble<FlatESLintConfig[]>[] = []

  configs.push(
    ignores(),
    imports(),
    comments(),
    jsonc(),
    sortKey(),
    javascript(),
    typescript(),
    vue(),
    yaml(),
    prettier() //Must be at the end of the array
  )

  const flatConfigs = await flat(...configs)

  return flatConfigs
}

export async function eslintConfigs(): Promise<FlatESLintConfig[]> {
  return await defineESlintConfig()
}
