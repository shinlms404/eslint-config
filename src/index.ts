import { defineESlintConfig } from './define'

export { defineESlintConfig }
