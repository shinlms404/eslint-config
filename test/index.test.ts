import { expect, test } from 'vitest'
import defineESlintConfig from '../dist/index'
import { getVueVersion } from '../src/utils'

test('define', async () => {
  const configs = defineESlintConfig()

  expect(configs).toMatchSnapshot()
})

test('vueVersion', async () => {
  const vueVersion = getVueVersion()

  expect(vueVersion).toBe(3)
})
