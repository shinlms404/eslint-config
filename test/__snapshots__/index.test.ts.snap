// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`define 1`] = `Promise {}`;
